.github/workflows/ci.yml

```diff
@@ -64,10 +76,29 @@ jobs:
           echo "npm-dir=$(npm config get cache)" >> $GITHUB_OUTPUT
       - uses: actions/cache@v4
         with:
-          path: ${{ steps.cache-dir.outputs.composer-dir }}
+          path: |
+            vendor
+            web/core
+            web/modules/contrib
+            web/themes/contrib
+            web/profiles/contrib
+            ${{ steps.cache-dir.outputs.composer-dir }}
           key: ${{ runner.os }}-composer-${{ hashFiles('composer.lock') }}
           restore-keys: |
             ${{ runner.os }}-composer-
+      - uses: actions/cache@v4
+        with:
+          path: |
+            **/node_modules
+            ${{ steps.cache-dir.outputs.npm-dir }}
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Cache Docker images
+        uses: ScribeMD/docker-cache@0.5.0
+        with:
+          key: ddev-images-${{ runner.os }}-${{ hashFiles('.ddev/config.yaml', '.ddev/docker-compose.cypress.yaml', '.ddev/docker-compose.redis.yaml') }}
```

Comment:
My concern with this key is that if there is a new DDEV version, what would be the impact here? I don't think the pipeline will fail but I am curious if the cache size will keep increasing indefinitely.

For example, we have DDEV 1.23.0 in cache and 1.23.1 is out. The key will still match and it will restore the images. But DDEV will download new images anyway.

Now, the question is what happens in the post action step. Will it see the new images and update the cache? If it updates the cache, will it remove the older images? If it doesn't remove the old images, then the cache size will increase significantly and it will take more time to restore the images than downloading fresh.

Reply:
My suggestion is to figure out if we can bring the DDEV version within this key. If that takes some time, then we should remove this into a separate PR.

Reply:
@hussainweb Thanks for noticing out concern with cache key. I have added `ddev version` to cache key. Please review.